PKHeX/Legality/Core.cs

```diff
@@ -927,7 +927,7 @@ internal static EncounterTrade getValidIngameTrade(PKM pkm, GameVersion gameSour
 
             if (!pkm.WasIngameTrade)
             {
-                if (pkm.HasOriginalMetLocation)
+                if (!pkm.HasOriginalMetLocation)
```

Comment:
A debug thinking, I will fix soon.
since at first for sm, it goes into this part and returned